tests/e2e/specs/shopper/checkout-different-addresses.test.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import {
+	merchant,
+	openDocumentSettingsSidebar,
+	uiUnblocked,
+} from '@woocommerce/e2e-utils';
+
+import {
+	visitBlockPage,
+	selectBlockByName,
+	saveOrPublish,
+} from '@woocommerce/blocks-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getNormalPagePermalink,
+	shopper,
+	visitPostOfType,
+} from '../../../utils';
+
+const config = require( 'config' );
+
+const block = {
+	name: 'Checkout',
+};
+
+const simpleProductName = '128GB USB Stick';
+const pageName = 'Checkout Block';
+
+if ( process.env.WOOCOMMERCE_BLOCKS_PHASE < 2 )
+	// eslint-disable-next-line jest/no-focused-tests
+	test.only( `skipping ${ block.name } tests`, () => {} );
+
+describe( `${ block.name } Block (frontend)`, () => {
+	let productPermalink;
+
+	beforeAll( async () => {
+		// prevent CartCheckoutCompatibilityNotice from appearing
+		await page.evaluate( () => {
+			localStorage.setItem(
+				'wc-blocks_dismissed_compatibility_notices',
+				'["checkout"]'
+			);
+		} );
+		await merchant.login();
+
+		// Display all address fields.
+		await visitBlockPage( pageName );
+		await openDocumentSettingsSidebar();
+		await selectBlockByName(
+			'woocommerce/checkout-shipping-address-block'
+		);
+		await expect( page ).toClick( 'label', { text: 'Company' } );
+		await expect( page ).toClick( 'label', {
+			text: 'Apartment, suite, etc.',
+		} );
+		await expect( page ).toClick( 'label', { text: 'Phone' } );
+		await saveOrPublish();
+
+		// Get product page permalink.
+		await visitPostOfType( simpleProductName, 'product' );
+		productPermalink = await getNormalPagePermalink();
+
+		await merchant.logout();
+	} );
+
+	afterAll( async () => {
+		// empty cart from shortcode page
+		await shopper.goToCart();
+		await shopper.removeFromCart( simpleProductName );
+		await page.evaluate( () => {
+			localStorage.removeItem(
+				'wc-blocks_dismissed_compatibility_notices'
+			);
+		} );
+	} );
+
+	it( 'allows customer to have different shipping and billing addresses', async () => {
+		await shopper.goToCheckoutBlock();
+		await page.goto( productPermalink );
+		await shopper.addToCart();
+		await shopper.goToCheckoutBlock();
+
+		await page.evaluate( () => {
+			const checkbox = document.querySelector(
+				'.wc-block-checkout__use-address-for-billing .wc-block-components-checkbox__input'
+			);
+			if ( checkbox.checked ) {
```

Comment:
Here we need to test if this checkbox is checked, as this is the default behaviour and we need to make sure this is respected.

After that a click needs to be made and then the fields checked.

Reply:
I am curious if we can test that this data is actually handled correctly when submitting the form. Because filling the form fields is a good thing, because it tests that all the fields are there. But we don't have an assertion that the data is used.

Would you mind trying to submit the form, and trying to see if:
- the form is successfully submitted 
- don't know if it would be possible to use `await page.waitForNavigation();` and search on the thank you page for the data under shipping address to match the data you introduced
<img width="714" alt="Screenshot 2022-02-16 at 13 44 06" src="https://user-images.githubusercontent.com/1628454/154267160-7d3efa60-53c6-4833-ac67-2aed12a7418f.png">
.

Reply:
Good point. The test should indeed check if all provided data appear on the order confirmation page as expected.

Reply:
```
		const checkbox = await page.$(
			'.wc-block-checkout__use-address-for-billing .wc-block-components-checkbox__input'
		);
		const isCheckBoxChecked = await (
			await checkbox.getProperty( 'checked' )
		 ).jsonValue();
		expect( isCheckBoxChecked ).toBe( true );
		await expect( page ).toClick(
			'.wc-block-checkout__use-address-for-billing .wc-block-components-checkbox__input'
		);
```